src/utils/dateUtils.js

```diff
@@ -0,0 +1,18 @@
+import dayjs from 'dayjs'
+import objectSupport from 'dayjs/plugin/objectSupport'
+import duration from 'dayjs/plugin/duration'
+dayjs.extend(objectSupport)
+dayjs.extend(duration)
+
+function calcDateDuration(date) {
+  const targetDate = dayjs().set({
+    days: date.day,
+    hours: date.hour,
+    minutes: date.minute,
+    seconds: date.second,
```

Comment:
you can change object structure:
```
{
  days: number,
  hours: number,
  minutes: number,
  seconds: number
}
```
So that to use object destruction in `set()` method:
e.g. `const targetDate = dayjs().set({...date})`